### ISSUE TITLE
`closing_brace` should not report on named parameters.

### ISSUE THREAD

**Issue by lhunath** [realm/SwiftLint#3686]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

```swift
let block = { self.layoutIfNeeded() } // Passes
UIView.animateWithDuration( 0.2, animations: block ) // Passes
UIView.animateWithDuration( 0.2, animations: { self.layoutIfNeeded() } ) // Violation
UIView.animateWithDuration( 0.2, animations: { self.layoutIfNeeded() }) // Passes
```

It's inconsistent for line 1 and 2 to be acceptable, but their combination to be unacceptable, while line 3, which is asymmetric, is somehow deemed more acceptable.

Ideally lines 1-3 should pass and line 4 should fail (or be ignored by this particular rule).

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint --path test.swift
Linting Swift files at paths test.swift
Linting 'test.swift' (1/1)
/Users/lhunath/workspace/lyndir/Spectre/ios/test.swift:3:70: warning: Closing Brace Spacing Violation: Closing brace with closing parenthesis should not have any whitespaces in the middle. (closing_brace)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.43.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Which Xcode version are you using (check `xcodebuild -version`)? Xcode 12.5.1 Build version 12E507

**Comment by PaulTaykalo** [comment]
This rule handles situations like `} )` only. Only closing braces are counted in.
So, I would say this works as expected.
https://realm.github.io/SwiftLint/closing_brace.html
